src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,688 @@
+            var methodSymbolInfo = (IMethodSymbol)document.SemanticModel.GetRequiredEnclosingSymbol(expression.SpanStart, cancellationToken);
+            var methodCallSites = await FindCallSitesAsync(document, methodSymbolInfo, cancellationToken).ConfigureAwait(false);
+
+            if (trampoline || overload)
+            {
+                return await RewriteSolutionWithNewMethodAsync(document,
+                expression, methodSymbolInfo, allOccurrences, parameterName, methodCallSites, trampoline, cancellationToken).ConfigureAwait(false);
+            }
+
+            return await RewriteSolutionAsync(document,
+                expression, methodCallSites, allOccurrences, parameterName, cancellationToken).ConfigureAwait(false);
+        }
+
+        protected static string GetNewParameterName(SemanticDocument document, TExpressionSyntax expression, CancellationToken cancellationToken)
+        {
+            if (ShouldRemoveVariableDeclaratorContainingExpression(document, expression, out var varDecl))
+            {
+                return varDecl;
+            }
+
+            var semanticFacts = document.Document.GetRequiredLanguageService<ISemanticFactsService>();
+            return semanticFacts.GenerateNameForExpression(document.SemanticModel, expression, capitalize: false, cancellationToken);
+        }
+
+        protected static bool ShouldRemoveVariableDeclaratorContainingExpression(
+            SemanticDocument document, TExpressionSyntax expression, out string varDecl)
+        {
+            var syntaxFacts = document.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var expressionDecl = expression.GetRequiredParent().GetRequiredParent();
+            varDecl = "";
+
+            if (syntaxFacts.IsVariableDeclarator(expressionDecl))
+            {
+                varDecl = syntaxFacts.GetIdentifierOfVariableDeclarator(expressionDecl).ValueText;
+                return true;
+            }
+
+            return false;
+        }
+
+        /// <summary>
+        /// Gets the method symbol the expression is enclosed within
+        /// </summary>
+        protected IMethodSymbol? GetMethodSymbolFromExpression(SemanticDocument semanticDocument,
+            TExpressionSyntax expression, CancellationToken cancellationToken)
+        {
+            var methodSymbol = (IMethodSymbol)semanticDocument.SemanticModel.GetRequiredEnclosingSymbol(expression.SpanStart, cancellationToken);
+            var syntaxFacts = semanticDocument.Document.GetRequiredLanguageService<ISyntaxFactsService>();
+            var methodDeclaration = expression.FirstAncestorOrSelf<SyntaxNode>(node => IsContainedInParameterizedDeclaration(node), ascendOutOfTrivia: true);
+            if (methodDeclaration is null)
+            {
+                return null;
+            }
+
+            return (IMethodSymbol)semanticDocument.SemanticModel.GetDeclaredSymbol(methodDeclaration, cancellationToken)!;
```

Comment:
i honestly don't get this metho.  you already called GetRequiredEnclosingSymbol... and you have that stored in `methodSymbol`.  But then you don't use `methodSymbol`... and you walk upwards to find the containing method, to get another method symbol...

Reply:
The method wasn't being used, I just forgot to remove it when I switched to using GetRequiredEnclosingSymbol